pkg/apis/nfd/v1alpha1/rule.go

```diff
@@ -30,13 +30,15 @@ import (
 // RuleOutput contains the output out rule execution.
 // +k8s:deepcopy-gen=false
 type RuleOutput struct {
-	Labels map[string]string
-	Vars   map[string]string
-	Taints []corev1.Taint
+	ExtendedResources []string
```

Comment:
Thanks for the PR @ArangoGutierrez. This is a great enhancement! Extended resources have a name and a value so in that sense this should 

```
	ExtendedResources map[string]string
```

or even

```
	ExtendedResources map[string]resource.Quantity
```

[ with `resource.Quantity` being the type from k8s apimachinery ]

OTOH, static quantity specified in the rule like 

```yaml
      extendedResources:
        - tdx.total_keys: 123
```

is not generally very useful so we need some dynamic solution. We already support templating for labels (and vars) so maybe following and extending that convention could be the solution. That is, being able to specify smth like this:

```yaml
      extendedResources:
        - tdx_keys: "{{ cpu.security.tdx.total_keys }}"
```

The stuff inside `{{ }}` could be fed to a real templating engine (what we do for `labelsTemplate` and `varsTemplate`) or then just do poor-man's stripping and splitting to figure out the feature value that the user wanted to expose. Thoughts @ArangoGutierrez @fidencio @mythi @zvonkok et al?

Reply:
Yeah I decided to change 
```go
	ExtendedResources map[string]string
```
to
```go
	ExtendedResources []string
```
to be able to define just as string, and then give the quantity and create the label. so we only take the name, and internally assign the value, that way we can 
```yaml
      extendedResources:
        - "cpu-security.tdx.total_keys"
```
So the `list` get's "enabled" if the rules match. 
That was my train of thought.  I do think `static quantity` is a bad approach, we should look for a solution like the provided in this PR or with templating.

Reply:
With this we really want to decouple extended resources from labels. I think we want the cabability to specify extended resource like `example.com/foo: 4` or `vendor.io/bar: "{{ cpu.security.tdx.total_keys }}"`

Also, we don't want to change the API from release to release (unless absolutely necessary) so we'd need to try to get it right so with that in mind `map[string]string`

Reply:
That's third item on top the list started here https://github.com/kubernetes-sigs/node-feature-discovery/pull/1099#discussion_r1153088587 :-)

Reply:
Got it, will work on the changes

Reply:
Just thought about the format, maybe we just go with simple @ to indicate value injection, like

```yaml
extendedResources:
  vendor.io/bar: @cpu.security.tdx.total_keys
```

`{{ }}` is a bit clumsy and possibly misleading if we're not doing real templating here. WDYT?